src/plugins/mer/meroptionswidget.cpp

```diff
@@ -249,60 +250,107 @@ void MerOptionsWidget::onSrcFolderApplyButtonClicked(const QString &newFolder)
     MerSdk *sdk = m_sdks[m_virtualMachine];
 
     if (newFolder == sdk->sharedSrcPath()) {
-        QMessageBox::information(this, tr("Choose a new folder"),
+        QMessageBox::information(this,
+                                 tr("Choose a new folder"),
                                  tr("The given folder (%1) is the current alternative source folder. "
                                     "Please choose another folder if you want to change it.").arg(sdk->sharedSrcPath()));
         return;
     }
 
-    if (!sdk->connection()->isVirtualMachineOff()) {
-        QPointer<QMessageBox> questionBox = new QMessageBox(QMessageBox::Question,
-                tr("Close Virtual Machine"),
-                tr("Close the \"%1\" virtual machine?").arg(m_virtualMachine),
-                QMessageBox::Yes | QMessageBox::No,
-                this);
-        questionBox->setInformativeText(
-                tr("Virtual machine must be closed before the source folder can be changed."));
-        if (questionBox->exec() != QMessageBox::Yes) {
-            // reset the path in the chooser
-            m_ui->sdkDetailsWidget->setSrcFolderChooserPath(sdk->sharedSrcPath());
-            return;
+    const bool vmIsOff = sdk->connection()->isVirtualMachineOff();
+    const bool vmWasClosed = !vmIsOff && showCloseVirtualMachineDialog(tr("Virtual machine must be closed before the source folder can be changed."));
```

Comment:
Shouldn't this do another `isVirtualMachineOff()` check after the dialog? `vmWasClosed` is actually misleading, it's true when the user clicked on the affirmative action on the dialog, no closing needs to be done?